test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlDocumentTest/ParseSectionIgnoresTagsInContentsOfScriptTag.stree.txt

```diff
@@ -1,52 +1,59 @@
-Markup block - Gen<None> - 53 - (0:0,0)
-    Markup span - Gen<Markup> - [] - SpanEditHandler;Accepts:Any - (0:0,0) - Tokens:1
-        SyntaxKind.Unknown;[];
-    Directive block - Gen<Directive:{section;RazorBlock;Unrestricted}> - 53 - (0:0,0)
-        Transition span - Gen<None> - [@] - SpanEditHandler;Accepts:None - (0:0,0) - Tokens:1
-            SyntaxKind.Transition;[@];
-        MetaCode span - Gen<None> - [section] - SpanEditHandler;Accepts:None - (1:0,1) - Tokens:1
-            SyntaxKind.Identifier;[section];
-        Code span - Gen<None> - [ ] - SpanEditHandler;Accepts:Whitespace - (8:0,8) - Tokens:1
-            SyntaxKind.Whitespace;[ ];
-        Code span - Gen<DirectiveToken {SectionName;Member;Opt:False}> - [Foo] - DirectiveTokenEditHandler;Accepts:NonWhitespace - (9:0,9) - Tokens:1
-            SyntaxKind.Identifier;[Foo];
-        Markup span - Gen<None> - [ ] - SpanEditHandler;Accepts:AllWhitespace - (12:0,12) - Tokens:1
-            SyntaxKind.Whitespace;[ ];
-        MetaCode span - Gen<None> - [{] - AutoCompleteEditHandler;Accepts:None,AutoComplete:[<null>];AtEnd - (13:0,13) - Tokens:1
-            SyntaxKind.LeftBrace;[{];
-        Markup block - Gen<None> - 38 - (14:0,14)
-            Markup span - Gen<Markup> - [ ] - SpanEditHandler;Accepts:Any - (14:0,14) - Tokens:1
-                SyntaxKind.Whitespace;[ ];
-            Tag block - Gen<None> - 8 - (15:0,15)
-                Markup span - Gen<Markup> - [<script>] - SpanEditHandler;Accepts:Any - (15:0,15) - Tokens:3
-                    SyntaxKind.OpenAngle;[<];
-                    SyntaxKind.Text;[script];
-                    SyntaxKind.CloseAngle;[>];
-            Markup span - Gen<Markup> - [foo<bar baz='] - SpanEditHandler;Accepts:Any - (23:0,23) - Tokens:7
-                SyntaxKind.Text;[foo];
-                SyntaxKind.OpenAngle;[<];
-                SyntaxKind.Text;[bar];
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[baz];
-                SyntaxKind.Equals;[=];
-                SyntaxKind.SingleQuote;['];
-            Expression block - Gen<Expr> - 4 - (36:0,36)
-                Transition span - Gen<None> - [@] - SpanEditHandler;Accepts:None - (36:0,36) - Tokens:1
+SyntaxKind.HtmlDocument - [0..53) - FullWidth: 53 - Slots: 1 - [@section Foo { <script>foo<bar baz='@boz'></script> }]
+    SyntaxKind.HtmlMarkupBlock - [0..53) - FullWidth: 53 - Slots: 1
+        SyntaxKind.HtmlTextLiteral - [0..0) - FullWidth: 0 - Slots: 1 - Gen<Markup> - SpanEditHandler;Accepts:Any
+            SyntaxKind.Unknown;[];
+        SyntaxKind.CSharpCodeBlock - [0..53) - FullWidth: 53 - Slots: 1
+            SyntaxKind.CSharpDirective - [0..53) - FullWidth: 53 - Slots: 2
```

Comment:
> It does. https://github.com/aspnet/Razor/blob/feature/razor-parser/src/Microsoft.AspNetCore.Razor.Language/Syntax/Syntax.xml#L168-L181
> One difference though is that I've combined 3 and 4 as just an optional body. I didn't think having that separate would provide much value. We can always add it later if we need it.

Ah, ya i'd definitely break that out at some point.

> Hmm. But directives aren't all metacode right? Most of the time, it contains csharp too. Right now anything that is named `RazorxyzSyntax` are all things that are either metacode or comments.

They aren't but they aren't C# either. They don't default to C# or Html, their structure is defined by a descriptor; I bring up this point because that's what differentiates `CodeBlock` from `MarkupBlock`.